### ISSUE TITLE
Why is the GPU usage low and the CPU usage high when training the model?

### ISSUE THREAD

**Issue by Assassintears** [ultralytics/yolov5#13349]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello, I train my data with Yolov5.7, it works well. But I find that the GPU usage is very low and the CPU usage  high, my train code like bellow:

```bash
python train.py --weights xxx --batch_size 8 --imgsz 832 --workers 4 --device 0
```

Well, if I set the workers parameter to 8 or bigger, the CPU usage may be high to 100%， the training process like bellow: 
![2](https://github.com/user-attachments/assets/dda8ec51-38d8-42a0-b95b-3b9b0b723079)

What should I do to get more high usage of GPU?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Assassintears, thank you for reaching out and sharing your concern with YOLOv5 🚀! This is an automated response, and an Ultralytics engineer will assist you soon.

To address your issue regarding GPU and CPU usage, it's crucial to ensure that your environment is set up optimally. 🛠️ Please provide a **minimum reproducible example** if possible, as this will greatly aid in diagnosing the issue.

In the meantime, here are some general pointers:

## General Tips

1. **Workers**: Adjusting the `--workers` parameter can affect how CPU resources are utilized. Finding the right balance based on your hardware is key. You might learn more about this in our [Training Tips](https://docs.ultralytics.com/guides/model-training-tips//).
2. **Batch size**: Try increasing your `--batch_size` if your GPU memory permits, as this can help improve GPU utilization.

## Requirements

Ensure you have [**Python>=3.8.0**](https://www.python.org/) and all necessary packages in [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed, including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):
```bash
git clone https://github.com/ultralytics/yolov5
cd yolov5
pip install -r requirements.txt
```

YOLOv5 performs best in environments with the appropriate dependencies including [CUDA](https://developer.nvidia.com/cuda) and [CUDNN](https://developer.nvidia.com/cudnn).

## Environments

Choose from several verified environments:
- **Notebooks**: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: [GCP Quickstart](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: [AWS Quickstart](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: [Docker Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status

Check if 🟢 <a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a> is green, indicating all CI tests are passing.

For a more advanced solution, consider trying out our new [YOLOv8](https://github.com/ultralytics/ultralytics) model, with enhanced performance and features:
```bash
pip install ultralytics
```

We appreciate your patience as we work to assist you further! 😊

**Comment by michael-mayo** [comment]
Try increasing your batch size. With the batch size you are using, GPU utilisation is only 2.79GB. How does performance change if you set the batch size to -1 (automatically set)?

**Comment by pderrenger** [comment]
Increasing the batch size can help improve GPU utilization. Try setting `--batch-size -1` to let the system automatically adjust it based on available memory. Let us know if this improves performance.
